lib/intent-models/server.js

```diff
@@ -0,0 +1,88 @@
+
+module.exports = {}; 
+const IntentBase = require('./base');
+const { IntentOperation } = require('./operation');
+const { IntentServerVariable } = require('./server-variable');
+const { IntentServerSecurity } = require('./server-security');
+
+/**
+ * 
+ * @class
+ * @alias module:@asyncapi/parser#IntentServer
+ * @extends IntentBase
+ */
+class IntentServer extends IntentBase {
+  /**
+   * @returns {boolean}
+   */
+  hasName() { return true; }
+  /**
+   * @returns {string|undefined}
+   */
+  name() { return 'some test name'; }
+
+  /**
+   * @returns {boolean}
+   */
+  hasProtocol() { return true; }
+  /**
+   * @returns {string|undefined}
+   */
+  protocol() { return 'test-protocol'; }
+
+  /**
+   * @returns {boolean}
+   */
+  hasUrl() { return true; }
+  /**
+   * @returns {string|undefined}
+   */
+  url() {return 'localhost';}
+
+  /**
+   * @returns {IntentOperation[]}
+   */
+  operations() {return [new IntentOperation()];}
```

Comment:
Should we include all the intents from the root in terms of the point of view for operations, such as `applicationPublishOperations` and `applicationSubscribeOperations`?

Reply:
I am not pretty sure if there is a clear use case. But perhaps makes sense. Could be v2.1.0 anyway so we don't have to add those now.

Reply:
Isn't the use case the same as the one for the root object? Just depend on which object you currently have access to 🤔